app/controllers/assignments_controller.rb

```diff
@@ -321,6 +321,8 @@ def is_int(val)
   end
 
   def set_questions_graders
+    # FIXME: This is way too complicated.
+
```

Comment:
Question files are complicated.  Getting them wrong is easy.  This code is essentially the parser for that data.  Your options are, either build a boatload of tiny classes that encapsulate each question and distribute the error checking across a bunch of files, or just have this code once and for all, and be done with it.  There's inherent complexity here that can't be wished away.